variables.tf

```
@@ -88,87 +103,133 @@ variable "postgres_public_access_cidrs" {
 }
 
 variable "default_nodepool_vm_type" {
-  default = "Standard_D8s_v4"
+  description = "The default virtual machine size for the Kubernetes agents"
+  type        = string
+  default     = "Standard_D8s_v4"
 }
+
 variable "kubernetes_version" {
   description = "The AKS cluster K8s version"
+  type        = string
   default     = "1.24"
 }
 
 variable "default_nodepool_max_nodes" {
   description = "(Required, when default_nodepool_auto_scaling=true) The maximum number of nodes which should exist in this Node Pool. If specified this must be between 1 and 100."
+  type        = number
   default     = 5
 }
+
 variable "default_nodepool_min_nodes" {
   description = "(Required, when default_nodepool_auto_scaling=true) The minimum number of nodes which should exist in this Node Pool. If specified this must be between 1 and 100."
+  type        = number
   default     = 1
 }
+
 variable "default_nodepool_os_disk_size" {
   description = "(Optional) The size of the OS Disk which should be used for each agent in the Node Pool. Changing this forces a new resource to be created."
+  type        = number
   default     = 128
 }
+
 variable "default_nodepool_max_pods" {
   description = "(Optional) The maximum number of pods that can run on each agent. Changing this forces a new resource to be created."
+  type        = number
   default     = 110
 }
 
 variable "default_nodepool_availability_zones" {
-  type    = list(string)
-  default = ["1"]
+  description = "A list of Availability Zones across which the Node Pool should be spread. Changing this forces a new resource to be created."
+  type        = list(string)
+  default     = ["1"]
 }
 
 # AKS advanced network config
 variable "aks_network_plugin" {
   description = "Network plugin to use for networking. Currently supported values are azure and kubenet. Changing this forces a new resource to be created."
   type        = string
   default     = "kubenet"
-  #TODO: add validation when value is 'azure'
+
+  validation {
+    condition     = contains(["kubenet", "azure"], var.aks_network_plugin)
+    error_message = "Error: Currently the supported values are `kubenet` and `azure`"
+  }
 }
 
 variable "aks_network_policy" {
-  description = "Sets up network policy to be used with Azure CNI. Network policy allows us to control the traffic flow between pods. Currently supported values are calico and azure. Changing this forces a new resource to be created."
+  description = "Sets up network policy to be used with Azure CNI. Network policy allows to control the traffic flow between pods. Currently supported values are calico and azure. Changing this forces a new resource to be created."
```

```suggestion
  description = "Sets up network policy to be used with Azure CNI. Network policy allows control of the traffic flow between pods. Currently supported values are calico and azure. Changing this forces a new resource to be created."
```
Consider asking Susan Pearsall to review all these new descriptions.